spec/support/request/web_helper.rb

```diff
@@ -100,46 +100,25 @@ def within_row(num, &block)
   end
 
   def select2_select(value, options)
-    id = options[:from]
-    options[:from] = "#s2id_#{id}"
-    targetted_select2(value, options)
-  end
+    open_select2("#s2id_#{options[:from]}")
 
-  def targetted_select2(value, options)
-    # find select2 element and click it
-    find(options[:from]).find('a').click
-    select_select2_result(value)
-  end
+    if options[:search]
+      page.find(:xpath, '//body')
+        .find(:css, '.select2-drop-active input.select2-input, .select2-dropdown-open input.select2-input')
+        .set(value)
+    end
 
-  def select_select2_result(value)
     page.find(:xpath, '//body')
-      .find(:css, '.select2-drop-active .select2-result-label', text: value)
+      .find(:css, '.select2-drop-active .select2-result-label', text: options[:select_text] || value)
       .click
   end
 
-  # Support having different texts to search for and to click in the select2
-  # field.
-  def targetted_select2_search(value, options)
-    page.execute_script %{$('#{options[:from]}').select2('open')}
-    page.execute_script "$('#{options[:dropdown_css]} input.select2-input').val('#{value}').trigger('keyup-change');"
-    select_select2_result(options[:select_text] || value)
-  end
-
-  def multi_select2_select(value, options)
-    find("#s2id_#{options[:from]}").find('ul li.select2-search-field').click
-    select_select2_result(value)
-  end
-
   def open_select2(selector)
-    page.execute_script "jQuery('#{selector}').select2('open');"
-  end
-
-  def close_select2(selector)
-    page.execute_script "jQuery('#{selector}').select2('close');"
+    page.find(selector).find(:css, '.select2-choice, .select2-search-field').click
   end
 
-  def set_select2_field(field, value)
-    page.execute_script %Q{$('#{field}').select2('val', '#{value}')}
+  def close_select2
+    page.find(:css, 'body').click
```

Comment:
is this not too generic? I dont understand why we dont need the selector...

Reply:
@luisramos0  Click outside the select2 container is enough to close it 🙂

Reply:
> can you add more details to the commit message or the PR about the change? 

Done!

Reply:
>  I'd recommend you use smaller commits that can help reviewers follow the different code modifications. this second commit is quite difficult to follow on the web_helper changes. for example, you could have one commit to replace execute_script, another one to merge methods (multi and targettted), etc.

Got it. Will do from now on.

> the build is broken, maybe another flaky spec?

Yeah 😞 

Do you mind restarting the build?

Reply:
>> Clicking outside the select2 container...

ah, ok, can you please add a comment to the method. I am sure other devs will alo miss that :-)

Reply:
Done.